x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -0,0 +1,580 @@
+- Users should be able to install and upgrade prebuilt rules on the `Basic` license and higher.
+- EPR is available for fetching the package unless explicitly indicated otherwise.
+- Only the latest **stable** package is checked for installation/upgrade and pre-release packages are ignored.
+
+### Non-functional requirements
+
+- Notifications, rule installation and rule upgrade workflows should work:
+  - regardless of the package type: with historical rule versions or without;
+  - regardless of the package registry availability: i.e., they should also work in air-gapped environments.
+- Rule installation and upgrade workflows should work with packages containing up to 15000 historical rule versions. This is the max number of versions of all rules in the package. This limit is enforced by Fleet.
+- Kibana should not crash with Out Of Memory exception during package installation.
+- For test purposes, it should be possible to use detection rules package versions lower than the latest.
+
+## Scenarios
+
+### Package installation
+
+#### **Scenario: Package is installed via Fleet**
+
+**Coverage**: 0 tests - covered by other scenarios. This scenario will be covered by the **Users install the latest prebuilt rules** scenario, which will include 1 e2e test which installs the rules from the real package.
+
+TODO: Where are scenarios for the real package?
+
+```Gherkin
+Given the package is not installed
+When user opens the Rule Management page
+Then the package gets installed in the background from EPR
+```
+
+#### **Scenario: Package is installed via bundled Fleet package in Kibana**
+
+**Coverage**: 1 integration test.
+
+```Gherkin
+Given the package is not installed
+And user is in an air-gapped environment
+When user opens the Rule Management page
+Then the package gets installed in the background from packages bundled into Kibana
+```
+
+#### **Scenario: Large package can be installed on a small Kibana instance**
+
+**Coverage**: 1 integration test.
+
+```Gherkin
+Given the package is not installed
+And the package contains the largest amount of historical rule versions (15000)
+And the Kibana instance has a memory heap size of X Mb (see note below)
+When user opens the Rule Management page
+Then the package is installed without Kibana crashing with an Out Of Memory error
+```
+
+TODO: Update the `heap size of X Mb` to a concrete value and update the note below.
+
+**Note**: The amount of memory is undefined as of now because, during implementation, we will try to find a memory heap threshold below which Kibana starts to crash constantly when you install the package with 15k rules. The plan is to then increase it to the point where it stops crashing, and use it as our value for this test.
+
+### Rule installation and upgrade via the Prebuilt rules API
+
+There's a legacy prebuilt rules API and a new one. Both should be tested against two types of the package: with and without historical rule versions.
```

Comment:
TODO: @banderror sync with @jpdjere

Reply:
Synced. @jpdjere will work on writing integration tests for the new endpoints according to this section. And we already have integration tests for the legacy endpoints that match this section.